### PR TITLE
ADE RHEL 8 Online Encryption Support

### PR DESCRIPTION
**Pull request by vermashi**
Checks if the new kernel params are supported. If they are, uses them; if they aren't it uses a custom dracut module to bridge the gap.

### CODE REVIEW

File: VMEncryption/main/DiskUtil.py [hunk 1]
```diff
@@ -227,20 +227,81 @@ def luks_add_cleartext_key(self, passphrase_file, dev_path, mapper_name, header_
...
+                if keyslot_segment and parts[0].strip().isnumeric():
+                    keyslot_lines.append(line)
+                    continue
+
+                if not parts[0][0].isspace():
+                    keyslot_segment = False
+            
+            return keyslot_lines
+
+    def luks_dump_keyslots(self, dev_path, header_file):
```
[the remark this answers is not in the document]
Comment (vermashi) [reply VMEncryption/main/DiskUtil.py:305]: Tested and it works as expected

File: VMEncryption/main/DiskUtil.py [hunk 2]
```diff
@@ -227,20 +227,81 @@ def luks_add_cleartext_key(self, passphrase_file, dev_path, mapper_name, header_
...
+                return splits[1]
+        return None
+
+    def _extract_luks_version_from_dump(self, luks_dump_out):
+        lines = luks_dump_out.split("\n")
+        for line in lines:
+            if "version:" in line.lower():
+                return line.split()[-1]
+
+    def _extract_luksv2_keyslot_lines(self, luks_dump_out):
```
[the remark this answers is not in the document]
Comment (vermashi) [reply VMEncryption/main/DiskUtil.py:255]: Added tests and comments

File: VMEncryption/main/EncryptionEnvironment.py [hunk 3]
```diff
@@ -28,6 +28,8 @@ def __init__(self, patching, logger):
         self.logger = logger
         self.encryption_config_path = '/var/lib/azure_disk_encryption_config/'
         self.daemon_lock_file_path = os.path.join(self.encryption_config_path, 'daemon_lock_file.lck')
+        self.resume_daemon_lock_file_path = os.path.join(self.encryption_config_path, 'resume_daemon_lock.lck')
```
[the remark this answers is not in the document]
Comment (vermashi) [reply VMEncryption/main/EncryptionEnvironment.py]: Checked, they are used.

**Review by ejarvi — APPROVED** [review]
LGTM

**Pull request merged**